app/assets/v2/js/pages/new_bounty.js

```diff
@@ -121,27 +121,57 @@ Vue.mixin({
 
       vm.form.keywords.push(item);
     },
-    validateBtc: function() {
+    validateFunderAddress: function() {
       let vm = this;
+      let isValid = true;
 
-      const ADDRESS_REGEX = new RegExp('^[13][a-km-zA-HJ-NP-Z1-9]{25,34}$');
-      const BECH32_REGEX = new RegExp('^bc1[ac-hj-np-zAC-HJ-NP-Z02-9]{11,71}$');
-      const valid_legacy = ADDRESS_REGEX.test(vm.form.funderAddress);
-      const valid_segwit = BECH32_REGEX.test(vm.form.funderAddress);
+      switch (vm.chainId) {
+        case '1995': {
+          // nervos
+          const ADDRESS_REGEX = new RegExp('^(ckb){1}[0-9a-zA-Z]{43,92}$');
+          const isNervosValid = ADDRESS_REGEX.test(vm.form.funderAddress);
+    
+          if (!isNervosValid && !vm.form.funderAddress.toLowerCase().startsWith('0x')) {
+            isValid = false;
+          }
+          break;
+        }
+
+        case '50797': {
+          // tezos
+          const ADDRESS_REGEX = new RegExp('^(tz1|tz2|tz3)[0-9a-zA-Z]{33}$');
+          const isTezosValid = ADDRESS_REGEX.test(vm.form.funderAddress);
+
+          if (!isTezosValid) {
+            isValid = false;
+          }
+          break;
+        }
+
+        case '0': {
+          // btc
+          const ADDRESS_REGEX = new RegExp('^[13][a-km-zA-HJ-NP-Z1-9]{25,34}$');
+          const BECH32_REGEX = new RegExp('^bc1[ac-hj-np-zAC-HJ-NP-Z02-9]{11,71}$');
+          const valid_legacy = ADDRESS_REGEX.test(vm.form.funderAddress);
+          const valid_segwit = BECH32_REGEX.test(vm.form.funderAddress);
+    
+          if (!valid_legacy && !valid_segwit) {
+            isValid = false;
+          }
+          break;
+        }
```

Comment:
^ Hmm since all these are being re-used for submit and bounty.
Would it make sense to move it to it's own file and invoking them ? It was alright duplicating when it was a few lines but might be cleaner to move it to it's own file and import. waddya think ?

Reply:
this change should be in a different